source/PlayerInfo.cpp

```diff
@@ -687,6 +687,28 @@ int64_t PlayerInfo::Salaries() const
 
 
 
+// Calculate the daily maintenance cost for all ships and in cargo outfits.
+int64_t PlayerInfo::Maintenance() const
+{
+	int64_t maintenance = 0;
+	for(const shared_ptr<Ship> &ship : ships)
+		if(!ship->IsDestroyed())
+		{
+			maintenance += ship->Attributes().Get("maintenance costs");
+			auto outfit = ship->Cargo().Outfits().begin();
+			for(unsigned int i = 0; i < ship->Cargo().Outfits().size(); ++i)
```

Comment:
I would expect the loop to loop over the actual iterator `while(outfit != ship->Cargo()->Outfits().end()){...}`.

Or alternatively (but the Endless Sky coding guidelines prefer the use of the iterator), that the outfit within the loop is retrieved using variable `i`.

Reply:
This should be a range-based for loop anyway, e.g. 
```c++
for( const auto &outfit : ...) {
  ...
```